plugins/modules/vmware_guest.py

```diff
@@ -2112,6 +2116,22 @@ def customize_vm(self, vm_obj):
             cc_mgr = self.content.customizationSpecManager
             if cc_mgr.DoesCustomizationSpecExist(name=custom_spec_name):
                 temp_spec = cc_mgr.GetCustomizationSpec(name=custom_spec_name)
+                spec_adapter_counter = 0
+                if len(temp_spec.spec.nicSettingMap) != len(self.params['networks']):
+                    self.module.fail_json(msg="Number of custom spec network adapter mappings does not equal the num. of defined network adapter mappings")
```

Comment:
I'm not sure if this is a good idea. I think there might be use cases where people don't want to configure _all_ NICs via a customization spec. Some people might want to configure the first NIC like this, but configure a second NIC somehow else.

Maybe the module should just fail if there's `prompt user` set somewhere in `customization_spec`. I mean, `prompt user` doesn't make much sense in an ansible module.

Just an idea, I don't really know what would be the best solution :-/

Reply:
Thanks for your Feedback @mariolenz, let me try to explain in a little more detail why in my opinion the behavior as it is today isn't optimal and we should consider some improvements.

The wording `prompt user` I used earlier might be misleading, I attached two Screenshots which makes it hopefully easier to understand.

![Screenshot1](https://user-images.githubusercontent.com/16312366/158034199-2c5075c8-7f18-4133-ace4-b6de925e0765.png)

![Screenshot2](https://user-images.githubusercontent.com/16312366/158034197-6cc38aa9-56f9-4e05-9570-ff60b79874d1.png)

Parameter `customization_spec` as it's implemented today, only works when either `DHCP` or `custom settings` is selected in Section Network of the VM Customization Specifications.
Using settings from parameter `networks` in combination with `customization_spec` just isn't possible because those will be ignored, same for the parameter `hostname`.

It will only work without VM Customization Specifications by defining `customization` and `networks` in the Playbook.

A use case would be, that I want to ensure similar settings across manual and automated deployments, by configuring Hostname as well as IP Settings individual per VM.
Therefore I want to use the same VM Customization Specification in both cases to avoid config drift between my Playbook and vSphere.

Another one is the fact that using a VM Customization Specification and be able to configure the interfaces by `networks` will reduce the amount of (duplicate) code in my Playbook, otherwise I need to add `customization` for every VM I deploy.

Regarding your concern:

> I'm not sure if this is a good idea. I think there might be use cases where people don't want to configure all NICs via a customization spec. Some people might want to configure the first NIC like this, but configure a second NIC somehow else.

Good and valid point, cases should be supported where the amount of Interfaces in parameter `networks` is lower than the VM Customization Specification.

In such a case we would assume that all other Interfaces will not be changed so they are set to `DHCP` or `static`, otherwise the Playbook will again fail.

Example:

- One Interfaces in `networks`, two in the `customization_spec` profile, settings of the first one will be set based on values in `networks`, independent what's configured in the VM Customization Specification, the second one will be left untouched.